### PR TITLE
Bug fixes and documentation

### PR DESCRIPTION
**Pull request by chkeita**
## Summary of the Pull Request

_What is this about?_

## PR Checklist
* [ ] Applies to work item: #xxx
* [ ] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/onefuzz) and sign the CLI.
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Info on Pull Request

_What does this include?_

## Validation Steps Performed

_How does someone test & validate?_

### CODE REVIEW

File: docs/unmnaged-nodes.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# Unmanaged Nodes
+The default mode of onefuzz is to run agent inside scaleset managed by the the Onefuzz instance. But it is possible to run outside of the Instance infrastructure.
```
Comment (Porges) [inline docs/unmnaged-nodes.md:2]:
```suggestion
The default mode of OneFuzz is to run the agents inside scalesets managed by the the Onefuzz instance. But it is possible to run outside of the Instance infrastructure.
```

File: docs/unmnaged-nodes.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+This is the unmanaged scenario. In this mode, the user can use its own resource to participate in the fuzzing.
```
Comment (Porges) [inline docs/unmnaged-nodes.md:3]:
```suggestion
This is the unmanaged scenario. In this mode, the user can use their own resource to participate in the fuzzing.
```

File: docs/unmnaged-nodes.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+## Set-up
+These are the steps to run an unmanaged node
+
+
+### Create an application registration in azure active directory
```
Comment (Porges) [inline docs/unmnaged-nodes.md:9]:
```suggestion
### Create an Application Registration in Azure Active Directory
```

File: docs/unmnaged-nodes.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+We will create the authentication method for the unmanaged node.
+From the [azure cli](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli) create a new **application registration**
```
Comment (Porges) [inline docs/unmnaged-nodes.md:11]:
```suggestion
From the [azure cli](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli) create a new **application registration**:
```

File: docs/unmnaged-nodes.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+```cmd
+az ad app create --display-name <registration_name>
+```
+Then use the application `app_id` in the result to create the associated **service principal**
```
Comment (Porges) [inline docs/unmnaged-nodes.md:15]:
```suggestion
Then use the application `app_id` in the result to create the associated **service principal**:
```

File: docs/unmnaged-nodes.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+```cmd
+az ad sp create --id <app_id>
+```
+Take note of the `id` returned by this request. We will call it the `principal_id`
+create an client_secret
```
Comment (Porges) [inline docs/unmnaged-nodes.md:21]:
```suggestion
Take note of the `id` returned by this request. We will call it the `principal_id`.

Next, create a `client_secret`:
```

File: docs/unmnaged-nodes.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+```
+az ad app credential reset --id <pp_id> --append
+```
+Take note of the `password` returned.
+
+### Authorize the application in onefuzz
```
Comment (Porges) [inline docs/unmnaged-nodes.md:28]:
```suggestion
### Authorize the application in OneFuzz
```

File: docs/unmnaged-nodes.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+From onefuzz deployment folder run the following script using the app_id
```
Comment (Porges) [inline docs/unmnaged-nodes.md:29]:
```suggestion
From the OneFuzz `deployment` folder run the following script using the `app_id` from above:
```

File: docs/unmnaged-nodes.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+```
+
+### Create an unmanaged pool
+From the onefuzz cli
+``` cmd
+onefuzz pools create <pool_name> <os> --unmanaged --object_id <principal_id>
+```
+
+### Download the agent binaries and the agent configuration
+Download a zip file containing the agent binaries
```
Comment (Porges) [inline docs/unmnaged-nodes.md:41]:
```suggestion
Download a zip file containing the agent binaries:
```

File: docs/unmnaged-nodes.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+```
+
+Under the client_credential section of the agent config file. update client_id and client
+```json
+{
+    "client_id": "<app_id>",
+    "client_secret": "<password>",
+}
+```
+Save the config to file.
```
Comment (Porges) [inline docs/unmnaged-nodes.md:60]:
```suggestion
Save the config to the file.
```

File: docs/unmnaged-nodes.md [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+
+### Start the agent.
+Navigate to folder corresponding to your os.
```
Comment (Porges) [inline docs/unmnaged-nodes.md:63]:
```suggestion
Navigate to the folder corresponding to your OS.
```

File: docs/unmnaged-nodes.md [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+Set the necessary environment variable by running the script set-env.ps1 of set-env.sh depending on your platform
```
Comment (Porges) [inline docs/unmnaged-nodes.md:64]:
```suggestion
Set the necessary environment variable by running the script `set-env.ps1` (for Windows) or `set-env.sh` (for Linux).
```

File: docs/unmnaged-nodes.md [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+Run the agent with the following command. If you need more nodes use different `machine_guid` for each
```
Comment (Porges) [inline docs/unmnaged-nodes.md:65]:
```suggestion
Run the agent with the following command. If you need more nodes use a different `machine_guid` for each one:
```

File: docs/unmnaged-nodes.md [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+```cmd
+onefuzz-agent run --machine_id <machine_guid> -c <path_to_config_file>
+```
+
+### Verify that the agent is registered to OneFuzz
+
+In the onefuzz cli run the following command
```
Comment (Porges) [inline docs/unmnaged-nodes.md:72]:
```suggestion
Using the OneFuzz CLI run the following command:
```

File: docs/unmnaged-nodes.md [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+
+```
+onefuzz nodes get <machine_guid>
+```
+
+This should return one entry. Verify that the `pool_name` matched the pool name created earlier
+From here you will be able to schedule jobs on that pool and they will be runnin
```
Comment (Porges) [inline docs/unmnaged-nodes.md:79]:
```suggestion
This should return one entry. Verify that the `pool_name` matched the pool name created earlier. 
From here you will be able to schedule jobs on that pool and they will be running.
```

**Review by Porges — COMMENTED** [review]
Mostly capitalization/punctuation changes to docs

File: src/runtime-tools/linux/set-env.sh [hunk 16]
```diff
@@ -0,0 +1,11 @@
+INSTANCE_OS_SETUP="/onefuzz/instance-specific-setup/linux/setup.sh"
+INSTANCE_SETUP="/onefuzz/instance-specific-setup/setup.sh"
+USER_SETUP="/onefuzz/setup/setup.sh"
+TASK_SETUP="/onefuzz/bin/task-setup.sh"
+MANAGED_SETUP="/onefuzz/bin/managed.sh"
+SCALESET_SETUP="/onefuzz/bin/scaleset-setup.sh"
+DOTNET_VERSIONS=('7.0.100' '6.0.403')
```
Comment (Porges) [inline src/runtime-tools/linux/set-env.sh:7]: Not sure that we need this here?

```suggestion
```

File: src/runtime-tools/win64/set-env.ps1 [hunk 17]
```diff
@@ -0,0 +1,12 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+$env:Path += ";C:\Program Files (x86)\Windows Kits\10\Debuggers\x64\;C:\onefuzz\win64;C:\onefuzz\tools\win64;C:\onefuzz\tools\win64\radamsa;$env:ProgramFiles\LLVM\bin"
+$env:ONEFUZZ_ROOT = "C:\onefuzz"
+$env:ONEFUZZ_TOOLS = "C:\onefuzz\tools"
+$env:LLVM_SYMBOLIZER_PATH = "C:\Program Files\LLVM\bin\llvm-symbolizer.exe"
+$env:RUST_LOG = "info"
+$env:DOTNET_VERSIONS = "7.0.100;6.0.403"
```
Comment (Porges) [inline src/runtime-tools/win64/set-env.ps1:9]: Also probably not needed?
```suggestion
```

File: src/runtime-tools/win64/set-env.ps1 [hunk 18]
```diff
@@ -0,0 +1,6 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+$env:Path += ";C:\Program Files (x86)\Windows Kits\10\Debuggers\x64\;C:\onefuzz\win64;C:\onefuzz\tools\win64;C:\onefuzz\tools\win64\radamsa;$env:ProgramFiles\LLVM\bin"
```
Comment (chkeita) [inline src/runtime-tools/win64/set-env.ps1:4]: Will need to be update to add dotnet dependency

File: src/runtime-tools/linux/set-env.sh [hunk 19]
```diff
@@ -0,0 +1,5 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+export DOTNET_ROOT=/onefuzz/tools/dotnet
+export DOTNET_CLI_HOME="$DOTNET_ROOT"
+export LLVM_SYMBOLIZER_PATH=/onefuzz/bin/llvm-symbolizer
```
Comment (Porges) [inline src/runtime-tools/linux/set-env.sh:5]: we set `RUST_LOG` on the Windows version. Should we set it here too?

**Review by Porges — APPROVED** [review]

**Pull request merged**